lib/Mojolicious/Sessions.pm

```diff
@@ -58,16 +59,13 @@ sub store {
     samesite => $self->samesite,
     secure   => $self->secure
   };
-  $c->signed_cookie($self->cookie_name, $value, $options);
+  my $method = $self->encrypted ? 'encrypted_cookie' : 'signed_cookie';
+  $c->$method($self->cookie_name, $value, $options);
 }
 
-sub _deserialize { Mojo::JSON::decode_json($_[0] =~ s/\}\KZ*$//r) }
+sub _deserialize { Mojo::JSON::decode_json($_[0]) }
```

Comment:
This will fail to deserialize existing signed sessions with the padding support removed

Reply:
Well spotted.